controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +134,335 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+						r.log.Info("Updating Secret", "Secret", secretTemplate.Name)
+						secretTemplate.Data = secret.Data
+						if err := r.Client.Update(r.ctx, secretTemplate); err != nil {
+							return reconcile.Result{}, fmt.Errorf("failed to update Secret: %v", err)
+						}
+					}
+				case "StorageClass":
+					var storageClassConfiguration StorageClassConfiguration
+					name := resource.Name
+					if name == "cephfs" {
+						storageClassConfiguration = r.newCephFilesystemStorageClassConfiguration(storageClassConfiguration)
+						enableRookCSICephFS = true
+					} else if name == "ceph-rbd" {
+						storageClassConfiguration = r.newCephBlockPoolStorageClassConfiguration(storageClassConfiguration)
+					}
+					for key, value := range resource.Data {
+						storageClassConfiguration.storageClass.Parameters[key] = value
+					}
+					availableStorageClassConfigurations = append(availableStorageClassConfigurations, storageClassConfiguration)
+				}
+			}
+			err = r.createStorageClasses(availableStorageClassConfigurations, storageClassClaimOwnerReference)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to create StorageClass: %v", err)
+			}
+			err = r.setRookCSICephFS(enableRookCSICephFS)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to set RookEnableCephFSCSIKey to EnableRookCSICephFS [%v]: %s", enableRookCSICephFS, err)
+			}
+			return reconcile.Result{}, nil
+		}
+
+		// Update the StorageClassClaim status.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimReady
+
+	}
+	// TODO(@rexagod): Deletion flow for SCC on Consumer side
+	// Close client-side connections.
+	providerClient.Close()
+
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
+func (r *StorageClassClaimReconciler) setRookCSICephFS(enableRookCSICephFS bool) error {
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	err := r.get(rookCephOperatorConfig)
+	if err != nil {
+		r.log.Error(err, "Unable to get RookCeph ConfigMap.", "RookCephConfigMap", klog.KRef(r.storageCluster.Namespace, "rook-ceph-operator-config"))
+		return err
+	}
+	enableDisableFlagStr := fmt.Sprintf("%v", enableRookCSICephFS)
+
+	// If the current state of 'ROOK_CSI_ENABLE_CEPHFS' flag is same, just return.
+	if rookCephOperatorConfig.Data["ROOK_CSI_ENABLE_CEPHFS"] == enableDisableFlagStr {
+		return nil
+	}
+	rookCephOperatorConfig.Data["ROOK_CSI_ENABLE_CEPHFS"] = enableDisableFlagStr
+	return r.Client.Update(r.ctx, rookCephOperatorConfig)
+}
+
+func (r *StorageClassClaimReconciler) createStorageClasses(availableStorageClassConfigurations []StorageClassConfiguration, storageClassClaimOwnerReference metav1.OwnerReference) error {
+	var skippedStorageClass []string
+	for _, storageclassConfiguration := range availableStorageClassConfigurations {
```

Comment:
Why does this util need to handle multiple storage classes at the same time? 
It is counter trivial, especially after the fact that you introduced "extra" code just to aggregate the different configurations into a list in the first place.

Please invoke this function with a single configuration when ever yot **need** to create a storage class